Fixes bug that never checked result of test in check_smoke.sh

jplehr opened the pull request:
Some tests were using the 'make run' command to encode a more complex command for testing. This would never trigger a test failure in check_smoke.sh as it relies on the 'make check' rule.

This patch fixes this for some tests.

I don't know if more tests in smoke show this behavior.

jplehr committed:
Fixes bug that never checked result of test

Some tests were using the 'make run' command to encode a more complex
command for testing. This would never trigger a test failure in
check_smoke.sh as it relies on the 'make check' rule.

This patch fixes this for some tests.

ronlieb reviewed (commented):
if you have this patch applied locally in aomp, could do a  ./run_rocm_test.sh 
want to make sure when this lands npsdb wont see any issues

jplehr commented:
I was going through all Makefiles that have their on `run` rule defined and added this change. This made some smoke tests fail again. 
I'd like @estewart08 to comment on the approach first anyway.

jplehr committed:
Modifies the Makefiles that had special `run` rules

- Let's the test cases aomp-issue376, collapse1, managed_memory, workgroup_size fail

jplehr commented:
I moved this to a `draft` so that it is not merged accidentally and breaking any CI.

estewart08 commented:
Yeah I like the idea, just need to make sure everyone is informed.

jplehr commented:
Any suggestions for better names? I find the current scheme somewhat bulky.

estewart08 commented:
Inside the test Makefile I might suggest changing the name:
```
TESTCMD -> RUNCMD
```
Then on the run target:
```
run:
  $(RUNCMD)
```

Makefiles.rules:
```
CHECK_COMMAND ?= $(or $(TESTCMD), ./$(TESTNAME) $(ARGS))
```
to
```
CHECK_COMMAND ?= $(or $(RUNCMD), ./$(TESTNAME) $(ARGS))
```

jplehr committed:
Rename as suggested

jplehr commented:
I think this PR requires first accepting these PRs, which all move tests back into smoke-fails.
Tests in 474 and 475 need to be looked at and see what is the expected output.
For 476, I have opened a Gerrit review that addresses this bug.

* https://github.com/ROCm-Developer-Tools/aomp/pull/474
* https://github.com/ROCm-Developer-Tools/aomp/pull/475
* https://github.com/ROCm-Developer-Tools/aomp/pull/476

jplehr commented:
Rebased to current `aomp-dev`, should be ready to merge.
